wallet/src/de/schildbach/wallet/service/BlockchainStateDataProvider.kt

```diff
@@ -173,7 +166,11 @@ class BlockchainStateDataProvider @Inject constructor(
                 }
 
                 val validMNsCount = if (mnlist.size() != 0) {
-                    mnlist.validMNsCount
+                    var virtualMNCount = 0
+                    mnlist.forEachMN(true) { entry ->
+                        virtualMNCount += if (entry.isHPMN) 4 else 1
+                    }
+                    virtualMNCount
```

Comment:
EvoNodes have 4x the collateral (4000 instead of 1000) and receive 4x times the block rewards, so we will count them as 4 regular masternodes.  

Each block only pays out a single masternode or 1/4 of a EvoNode.  EvoNodes receive 4 blocks of rewards in a row.  APY calculations are based on a single block.

Without this fix, we are using the number of actual masternode servers which will inflate the APY by around 10%.

Future improvements:
1.  We could base the APY on a number of blocks in recent history such as the past 100.
2.  Masternode reward calculations could be moved into dashj, but not sure which class would get the functions (Block or Masternode List)